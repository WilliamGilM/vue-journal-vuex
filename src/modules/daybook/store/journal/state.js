

export default () => ({
  isLoading: true,
  entries: [
    {
      id: new Date().getTime(),
      date: new Date().toDateString(),
      text: 'Ex reprehenderit incididunt in officia incididunt voluptate incididunt.',
      picture: null,
    },
    {
      id: new Date().getTime() + 1000,
      date: new Date().toDateString(),
      text: 'Duis pariatur exercitation ad do sunt ut commodo nostrud aute commodo esse qui laboris.',
      picture: null,
    },
    {
      id: new Date().getTime() + 2000,
      date: new Date().toDateString(),
      text: 'Irure nisi dolor sint commodo.',
      picture: null,
    },
  ]
})