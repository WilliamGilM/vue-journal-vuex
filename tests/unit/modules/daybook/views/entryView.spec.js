import { shallowMount } from "@vue/test-utils";
import { createStore } from "vuex";

import Swal from 'sweetalert2'

import journal from "@/modules/daybook/store/journal";
import { journalState } from "../../../mock-data/test-journal-state";
import EntryView from "@/modules/daybook/views/EntryView";
import { nextTick } from "vue";

const createVuexStore = (initialState) =>
  createStore({
    modules: {
      journal: {
        ...journal,
        state: { ...initialState },
      },
    },
  })

  // espia de api sweetalert2
  jest.mock('sweetalert2', () => ({
    fire: jest.fn(),
    showLoading: jest.fn(),
    close: jest.fn()
  }))

describe("Pruebas en el EntryView", () => {
  const store = createVuexStore(journalState)
  // Se tiene el control absoluto del dispatch en las pruebas de jest
  store.dispatch = jest.fn()

  const mockRouter = {
    push: jest.fn(),
  };

  let wrapper;

  beforeEach(() => {
    // limpiar los mocks
    jest.clearAllMocks();
    wrapper = shallowMount(EntryView, {
      props: {
        id: "-Nbrdj3H7vX-k9Dkp_Bd",
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    })
  })

  test("debe de sacar al usuario porque el id no existe", () => {
    const wrapper = shallowMount(EntryView, {
      props: {
        id: "Este ID no existe en el store",
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    })

    expect(mockRouter.push).toHaveBeenCalledWith({ name: "no-entry" })

  });

  test("debe de mostrar la entrada correctamente", () => {

    expect(wrapper.html()).toMatchSnapshot()
    expect(mockRouter.push).not.toHaveBeenCalled()

  })

  test('debe de borrar la entrada y salir', (done) => {

    Swal.fire.mockReturnValueOnce( Promise.resolve({ isConfirmed: true}) )

    wrapper.find('.btn-danger').trigger('click')

    expect( Swal.fire ).toHaveBeenCalledWith({
      title: '¿Está seguro?',
      text: 'Una vez borrado, no se puede recuperar',
      showDenyButton: true,
      confirmButtonText: 'Si, estoy seguro'
    })

 
    setTimeout(() => {

      expect( store.dispatch ).toHaveBeenCalledWith('journal/deleteEntry', '-Nbrdj3H7vX-k9Dkp_Bd')
      expect( mockRouter.push ).toHaveBeenCalled()
      done()

    }, 1);

    /* Swal.fire.mockReturnValueOnce(Promise.resolve({ isConfirmed: true }))
        await wrapper.find('.btn-danger').trigger('click')
        expect(Swal.fire).toHaveBeenCalledWith({
            title: '¿Estas seguro?',
            text: 'Se borrara y lo perderas para siempre, eso es mucho tiempo',
            showDenyButton: true,
            denyButtonText: 'No, aún no',
            icon: 'question',
            confirmButtonText: 'Sin miedo al éxito'
        })
        expect(mockRouter.push).toHaveBeenCalled()
        expect(store.dispatch).toHaveBeenCalledWith('journal/deleteEntry', '-MhOjWw86ptCQT1oGvov') */
    
    
    
  })


})
