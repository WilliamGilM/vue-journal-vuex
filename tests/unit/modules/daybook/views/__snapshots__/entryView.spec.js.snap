// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el EntryView debe de mostrar la entrada correctamente 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3 fw-bold">15</span><span class="mx-1 fs-3">Agosto</span><span class="mx-2 fs-4 fw-light">2023, Martes</span></div>
  <div><input type="file" accept="image/png, image/jpeg" style="display: none;"><button class="btn btn-danger mx-2"> Borrar <i class="fa fa-trash-alt"></i></button><button class="btn btn-primary"> Subir foto <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea placeholder="¿Qué sucedió hoy?"></textarea></div>
<img src="https://res.cloudinary.com/dsbmzflnt/image/upload/v1692077672/rp2w9ekxqscslti2hgfj.jpg" alt="entry-picture" class="img-thumbnail">
<!--v-if-->
<fab-stub icon="fa-save"></fab-stub>
`;
